examples/11_hens/ensemble_utilities.py

```diff
@@ -0,0 +1,716 @@
+        """
+        xx2, coords2 = map_coords(xx, coords, self.cyclone_tracking_ic)
+        rt = coords2["time"][0]  # run time/initialisation time
+        lt = coords2["lead_time"][0]  # lead time
+        vt = rt + lt  # verification time
+
+        # track_element_dict["time_list"].append(vt)
+
+        coords2["time"] = vt
+
+        track_value, _ = self.cyclone_tracking(xx2, coords2)
+        # del track_coords["lead_time"]
+        # track_value = track_value.squeeze(2)
+        # track_element_dict["track_element_list"].append(track_value)
+        # track_element_dict["track_coords_list"].append(track_coords)
+        # return track_element_dict
+        return track_value
+
+    def combine_track_elements(
+        self,
+        track_element_dict: dict,
+    ) -> tuple[torch.Tensor, OrderedDict, np.ndarray[np.int64]]:
+        """
+        Combines individual track elements (representing a time step) into a large tensor representing all timesteps.
+
+        Parameters
+        ----------
+        track_element_dict : dict
+            Dictionary containing lists for TC center candidates, their coordinates, and verification times.
+            It should have the following keys:
+            - 'track_element_list': List of torch.Tensor objects containing TC center candidates for each timestep.
+            - 'track_coords_list': List of CoordSystem objects corresponding to each element in 'track_element_list'.
+            - 'time_list': List of np.datetime64 objects representing the verification times for each element in 'track_element_list'.
+
+        Returns
+        -------
+        tuple[torch.Tensor, OrderedDict, np.ndarray[np.int64]]
+            - out_tensor: torch.Tensor containing TC center candidates for all timesteps.
+            - track_coords_final: OrderedDict representing the final coordinate system.
+            - member_ids: np.ndarray[np.int64] containing ensemble member IDs.
+        """
+        track_element_list = track_element_dict["track_element_list"]
+        track_coords_list = track_element_dict["track_coords_list"]
+        time_list = track_element_dict["time_list"]
+        out_tensor = torch.nested.nested_tensor(
+            track_element_list, dtype=torch.float32, device=track_element_list[0].device
+        )
+        out_tensor = torch.swapaxes(
+            torch.nested.to_padded_tensor(out_tensor, torch.nan), 0, 2
+        )
+        # remove lead time dim
+        out_tensor = out_tensor.squeeze(0)
+
+        track_coords_final = track_coords_list[-1]
+        member_ids = track_coords_final["ensemble"]
+        del track_coords_final["ensemble"]
+        track_coords_final["time"] = time_list
+        track_coords_final["point"] = np.sort(
+            np.unique(np.concatenate([x["point"] for x in track_coords_list]))
+        )
+        return out_tensor, track_coords_final, member_ids
+
+    def connect_centres_to_tracks(
```

Comment:
Moving a cyclone related items into a hens_cyclone_utils.py would perhaps be good